at-loader没有抛出错误

### 遇到的问题
使用umi+ts组合时，ts/tsx中的类型错误并不会被at-loader报出来。比如以下代码，可以被正常的编译运行
```ts
const a: number = '1'
```

### 原因
af-webpack中默认开启了at-loader的transpileOnly属性
https://github.com/umijs/umi/blob/9a29b3202d89e78a1967ab4d417f298637c3ed52/packages/af-webpack/src/getConfig.js#L460
https://github.com/s-panferov/awesome-typescript-loader#transpileonly-boolean

### 建议
af-webpack中不再指定at-loader的参数，由用户在tsconfig.json中指定

### 其他问题
虽然af-webpack中已经给at-loader设定了`exclude: /node_modules/`([ref](https://github.com/umijs/umi/blob/9a29b3202d89e78a1967ab4d417f298637c3ed52/packages/af-webpack/src/getConfig.js#L452))，但是at-loader依然会把node_modules中的错误报出来，比如
```
Failed to compile.

[at-loader] ./node_modules/antd-mobile/lib/modal/PropsType.d.ts:3:38
    TS2307: Cannot find module 'react-native'.
```
这种情况有人在at-loader里提过issue [awesome-typescript-loader#492](https://github.com/s-panferov/awesome-typescript-loader/issues/492)
解决方案是指定at-loader的[reportFiles参数](https://github.com/s-panferov/awesome-typescript-loader#reportfiles-string)

### 临时解决方案
step1: 将项目文件夹内`node_modules/af-webpack/lib/getConfig.js`第418行删掉
```diff
  configFileName: opts.tsConfigFile || (0, _path.join)(opts.cwd, 'tsconfig.json'),
- transpileOnly: true
```
step2: 在tsconfig.json中添加如下at-loader配置
```json
"awesomeTypescriptLoaderOptions": {
  "reportFiles": ["src/**/*.{ts,tsx}"]
}
```

--- a/packages/af-webpack/src/getConfig.js
+++ b/packages/af-webpack/src/getConfig.js
@@ -480,6 +480,7 @@ export default function getConfig(opts = {}) {
                 configFileName:
                   opts.tsConfigFile || join(opts.cwd, 'tsconfig.json'),
                 transpileOnly: true,
+                ...(opts.typescript || {}),
               },
             },
           ],

--- a/packages/af-webpack/src/getUserConfig/configs/typescript.js
+++ b/packages/af-webpack/src/getUserConfig/configs/typescript.js
@@ -1,0 +1,14 @@
+import assert from 'assert';
+import isPlainObject from 'is-plain-object';
+
+export default function() {
+  return {
+    name: 'typescript',
+    validate(val) {
+      assert(
+        isPlainObject(val),
+        `The typescript config must be Plain Object, but got ${val}`,
+      );
+    },
+  };
+}

